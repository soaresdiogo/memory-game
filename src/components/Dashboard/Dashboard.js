import React from 'react';
import Grid from '@material-ui/core/Grid';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

const styles = {
  root: {
    height: '100vh',
    backgroundColor: '#ccc',
  },
};

const Dashboard = ({ children, classes }) => (
  <Grid
    data-testid="dashboard"
    className={classes.root}
    alignItems="center"
    justify="center"
    container
  >
    {children}
  </Grid>
);

Dashboard.defaultProp = {
  classes: PropTypes.node,
};

Dashboard.propTypes = {
  children: PropTypes.node.isRequired,
};

export default withStyles(styles)(Dashboard);
