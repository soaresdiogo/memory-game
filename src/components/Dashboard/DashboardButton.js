import React from 'react';
import Button from '@material-ui/core/Button';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const DashboardButton = ({ children, to }) => (
  <Button
    data-testid="dashboardButton"
    component={Link}
    to={to}
    variant="outlined"
    color="primary"
    size="large"
    fullWidth
  >
    {children}
  </Button>
);

DashboardButton.defaultProp = {
  to: PropTypes.string,
};

DashboardButton.propTypes = {
  children: PropTypes.string.isRequired,
};

export default DashboardButton;
