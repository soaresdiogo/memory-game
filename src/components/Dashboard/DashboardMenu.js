import React from 'react';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';

const styles = (theme) => ({
  root: {
    padding: theme.spacing(6),
  },
  title: {
    paddingBottom: theme.spacing(6),
  },
});

const DashboardMenu = ({ children, classes, title }) => (
  <Paper data-testid="dashboardMenu" className={classes.root}>
    <Typography
      data-testid="dashboardMenuTitle"
      variant="h4"
      align="center"
      className={classes.title}
    >
      {title}
    </Typography>
    {children}
  </Paper>
);

DashboardMenu.defaultProp = {
  classes: PropTypes.node,
  children: PropTypes.node,
};

DashboardMenu.propTypes = {
  title: PropTypes.string.isRequired,
};

export default withStyles(styles)(DashboardMenu);
