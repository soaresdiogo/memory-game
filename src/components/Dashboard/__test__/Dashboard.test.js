import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Dashboard from '../Dashboard';

import '@testing-library/jest-dom/extend-expect';

const DashboardComponent = <Dashboard>Content</Dashboard>;

describe('Dashboard component', () => {
  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(DashboardComponent, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('renders div correctly', () => {
    const { getByTestId } = render(DashboardComponent);
    expect(getByTestId('dashboard')).toBeTruthy();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(DashboardComponent).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
