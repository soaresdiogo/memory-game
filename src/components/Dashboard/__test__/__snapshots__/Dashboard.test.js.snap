// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component matches snapshot 1`] = `
<div
  className="MuiGrid-root Dashboard-root-3 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  data-testid="dashboard"
>
  Content
</div>
`;
